Gp 112 fe navbar for public pages

thuanvmse161224 opened the pull request.

thuanvmse161224 committed:
feat: navbar wip
- todo:  navItems to works at other pages other than root

thuanvmse161224 committed:
refactor: Changing file and function name

thuanvmse161224 committed:
fix: fixing forgot bg bug
- Calculate the navbar height
- Then calculate appropriate height and pos for img (absolute)

thuanvmse161224 committed:
fix: forgot bg bug
- no more style err when :
   + refresh
   + access from url
   + access from routing (react-router)

danielNg91 commented on src/Layout/GuestLayout/Navbar/Navbar.tsx:
```diff
@@ -0,0 +1,52 @@
+import * as styles from './styles'
+
+
+const navigationItems = [
```
add type for this const

thuanvmse161224 replied:
working on it

thuanvmse161224 replied:
done

danielNg91 commented on src/Layout/GuestLayout/Navbar/styles.ts:
```diff
@@ -0,0 +1,43 @@
+	color: 'white',
+};
+
+export const textLogoStyle = { 
```
add type, use Object.freeze for all const in this file

thuanvmse161224 replied:
done

thuanvmse161224 committed:
refactor: add const freeze, add type
- add types for navItems data
- move type to types.ts
- const freezing in styles.ts
- add type for sx props in textLogoStyle, like mui-Typography

danielNg91 commented on src/Layout/GuestLayout/Navbar/styles.ts:
```diff
@@ -0,0 +1,50 @@
+import { CSSProperties } from 'react';
+
+
+export const navbarStyle: CSSProperties = Object.freeze({
```
it is generic function, use Object.freeze<CSSProperties> to get advantage of type suggestion

thuanvmse161224 replied:
done

thuanvmse161224 committed:
refactor: styles const refactor
- Refactor with Object.freeze generic
- with CSSProperties & SxProps<Theme>
- also extract styles in login, contact, register
